tests/unit/test_athena.py

```diff
@@ -203,6 +214,7 @@ def test_athena_create_ctas(path, glue_table, glue_table2, glue_database, glue_c
     ensure_athena_ctas_table(ctas_query_info=ctas_query_info, boto3_session=boto3_session)
 
 
+@pytest.mark.xfail(is_ray_modin, raises=AssertionError, reason="Index equality regression")
```

Comment:
So both casting to pandas for equality checks and xfailing are needed in this test? Could you put what is different about these data frames after you cast them that fails the assertions into the xfail reason instead of the default message, please?

Reply:
Also, just to confirm - are you using `pandas_equals` only where it's necessary i.e. where modin equality doesn't work (such as [4071](https://github.com/modin-project/modin/issues/4071), [5158](https://github.com/modin-project/modin/issues/5158)) or you've replaced all df.equals?

Reply:
Hmmm I'm not using `pandas_equals` in his particular test case.

I basically have to use `pandas_equals` everywhere that we potentially compare Modin Data Frames. Otherwise the equality comparison fails in every test case where we compare data frames or data series.

Reply:
Right, sorry, the diff got me confused. 

Would be great to track down what exactly causes the inequality and raise this with modin team.